packages/client/lib/sync/skeleton.ts

```diff
@@ -652,6 +656,10 @@ export class Skeleton extends MetaDBManager {
         numBlocksInserted = await this.chain.putBlocks([block], true)
       } catch (e) {
         this.config.logger.error(`fillCanonicalChain putBlock error=${(e as Error).message}`)
+        if (oldHead !== null) {
```

Comment:
```suggestion
        if (oldHead !== null && oldHead >= block.header.number) {
```
should this also be done?